packages/commonwealth/server.ts

```diff
@@ -261,43 +261,47 @@ async function main() {
     // TODO: this requires an immediate response if in production
   }
 
-  if (!NO_TOKEN_BALANCE_CACHE) await tokenBalanceCache.start();
-  if (!NO_RULE_CACHE) await ruleCache.start();
-  const banCache = new BanCache(models);
-  const globalActivityCache = new GlobalActivityCache(models);
-
-  // TODO: should we await this? it will block server startup -- but not a big deal locally
-  if (!NO_GLOBAL_ACTIVITY_CACHE) await globalActivityCache.start();
-
-  // Declare Validation Middleware Service
-  // middleware to use for all requests
-  const dbValidationService: DatabaseValidationService =
-    new DatabaseValidationService(models);
-
-  setupAPI(
-    '/api',
-    app,
-    models,
-    viewCountCache,
-    tokenBalanceCache,
-    ruleCache,
-    banCache,
-    globalActivityCache,
-    dbValidationService
-  );
-
-  // new API
-  addExternalRoutes('/external', app, models, tokenBalanceCache);
-  addSwagger('/docs', app);
-
-  setupCosmosProxy(app, models);
-  setupIpfsProxy(app);
-  setupEntityProxy(app);
-  setupAppRoutes(app, models, devMiddleware, templateFile, sendFile);
-
-  setupErrorHandlers(app, rollbar);
-
-  setupServer(app, rollbar, models, rabbitMQController);
+  try {
+    if (!NO_TOKEN_BALANCE_CACHE) await tokenBalanceCache.start();
+    if (!NO_RULE_CACHE) await ruleCache.start();
+    const banCache = new BanCache(models);
+    const globalActivityCache = new GlobalActivityCache(models);
+
+    // TODO: should we await this? it will block server startup -- but not a big deal locally
+    if (!NO_GLOBAL_ACTIVITY_CACHE) await globalActivityCache.start();
+
+    // Declare Validation Middleware Service
+    // middleware to use for all requests
+    const dbValidationService: DatabaseValidationService =
+      new DatabaseValidationService(models);
+
+    setupAPI(
+      '/api',
+      app,
+      models,
+      viewCountCache,
+      tokenBalanceCache,
+      ruleCache,
+      banCache,
+      globalActivityCache,
+      dbValidationService
+    );
+
+    // new API
+    addExternalRoutes('/external', app, models, tokenBalanceCache);
+    addSwagger('/docs', app);
+
+    setupCosmosProxy(app, models);
+    setupIpfsProxy(app);
+    setupEntityProxy(app);
+    setupAppRoutes(app, models, devMiddleware, templateFile, sendFile);
+
+    setupErrorHandlers(app, rollbar);
+
+    setupServer(app, rollbar, models, rabbitMQController);
+  } catch (e) {
+    console.log(e);
```

Comment:
Can we leave an error with traceback and also exit if we get an error during server startup?